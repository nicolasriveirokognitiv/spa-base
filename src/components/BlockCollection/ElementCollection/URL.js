import React from 'react';
import useTestMode from 'Hooks/useTestMode';
import { Input } from 'antd';
import { toolPropTypes } from 'Constants/propTypes';
import { elementCategories } from 'Constants/global';

function URL({ onChange, value, placeholder, defaultValue, testId }) {
    const testProps = useTestMode(testId);
    const handleChange = e => onChange(e.target.value);

    return (
        <Input
            addonBefore="https://"
            value={value}
            defaultValue={defaultValue}
            placeholder={placeholder}
            onChange={handleChange}
            {...testProps}
        />
    );
}

URL.propTypes = toolPropTypes;
URL.defaultProps = {
    wrapperClassName : 'type-url',
    label            : '',
    value            : null,
    onChange         : () => {},
    propertyKey      : 'url',
    category         : elementCategories.CONTENT,
};

export default URL;
