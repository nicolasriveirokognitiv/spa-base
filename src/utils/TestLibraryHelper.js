/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import initStore from 'State/store';

const store = initStore();

const renderWithRedux = children => render(
    <Provider store={store}>
        {children}
    </Provider>,
);

export * from '@testing-library/react';
export { renderWithRedux, render };
