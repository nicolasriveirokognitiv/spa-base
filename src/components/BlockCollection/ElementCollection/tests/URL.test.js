import React from 'react';
import { render } from 'Utils/TestLibraryHelper';
import EditorContext from 'State/editorContext';
import URL from '../URL';

describe('URL', () => {
    let container;

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
    });

    const renderURL = props => render(
        <EditorContext.Provider value={{ isTestModeActive: true }}>
            <URL {...props} />
        </EditorContext.Provider>,
    );

    test('Should render Correctly', () => {
        expect(<URL />).toBeDefined();
    });

    test('Should match snapshot', async () => {
        const { asFragment } = render(<URL />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('URL should be recibed on props', () => {
        const { getByTestId } = renderURL({ value: 'URL', testId: 'url-input' });
        const elem = getByTestId('url-input');
        expect(elem.value).toBe('URL');
    });
});
