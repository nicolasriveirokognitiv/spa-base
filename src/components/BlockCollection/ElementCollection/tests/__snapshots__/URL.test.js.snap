// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`URL Should match snapshot 1`] = `
<DocumentFragment>
  <span
    class="ant-input-group-wrapper"
  >
    <span
      class="ant-input-wrapper ant-input-group"
    >
      <span
        class="ant-input-group-addon"
      >
        https://
      </span>
      <input
        class="ant-input"
        type="text"
        value=""
      />
    </span>
  </span>
</DocumentFragment>
`;
